Go client : MapStructMarshal : call of reflect.Value.Type on zero Value

The MapStructMarshal function cannot handle a not valid reflect.Value (nil or invalid)
This test function will fail due to the Z:nil value..
```
func TestMapStructMarshal(t *testing.T) {
	type Z struct{}
	type X struct {
		Z *Z
	}

	x := X{Z: nil}
	valueX := reflect.ValueOf(x)
	mymap := MapStructMarshal(valueX, "tag string", true, true)
	t.Log(mymap)

}

```

I propose you add something like this on row 41 in structmap.go
```
if !value.IsValid() {
		return m
	}
```

With that the test passes.

<details><summary>Full Stacktrace</summary>
 <p>

```
panic: reflect: call of reflect.Value.Type on zero Value

goroutine 82 [running]:
reflect.Value.Type(0x0, 0x0, 0x0, 0x1a9e500, 0xc000113770)
        /usr/local/go/src/reflect/value.go:1872 +0x183
github.com/zeebe-io/zeebe/clients/go/internal/utils.MapStructMarshal(0x0, 0x0, 0x0, 0x195e720, 0x4, 0x100, 0x2c)
        ..../github.com/zeebe-io/zeebe/clients/go/internal/utils/structmap.go:41 +0x6e
github.com/zeebe-io/zeebe/clients/go/internal/utils.MapValueMarshal(0x18fc340, 0xc0000f4180, 0x196, 0x195e720, 0x4, 0xc000000100, 0x196, 0x18fc340)
        ..../github.com/zeebe-io/zeebe/clients/go/internal/utils/structmap.go:102 +0x101
github.com/zeebe-io/zeebe/clients/go/internal/utils.MapStructMarshal(0x19323e0, 0xc0000f4160, 0x199, 0x195e720, 0x4, 0x100, 0xc0001eb3d8)
        ..../github.com/zeebe-io/zeebe/clients/go/internal/utils/structmap.go:80 +0x2f4
github.com/zeebe-io/zeebe/clients/go/internal/utils.MapValueMarshal(0x192bba0, 0xc000075220, 0x196, 0x195e720, 0x4, 0xc000000100, 0x196, 0x192bba0)
        ..../github.com/zeebe-io/zeebe/clients/go/internal/utils/structmap.go:102 +0x101
github.com/zeebe-io/zeebe/clients/go/internal/utils.MapStructMarshal(0x193e220, 0xc000075220, 0x199, 0x195e720, 0x4, 0x100, 0x2c)
        ..../github.com/zeebe-io/zeebe/clients/go/internal/utils/structmap.go:69 +0x4e9
github.com/zeebe-io/zeebe/clients/go/internal/utils.MapValueMarshal(0x19309a0, 0xc0000e40d0, 0x196, 0x195e720, 0x4, 0xc000000100, 0x196, 0x19309a0)
        ..../github.com/zeebe-io/zeebe/clients/go/internal/utils/structmap.go:102 +0x101
github.com/zeebe-io/zeebe/clients/go/internal/utils.MapStructMarshal(0x1877e60, 0xc0000e40d0, 0x199, 0x195e720, 0x4, 0x100, 0x1897c00)
        ..../github.com/zeebe-io/zeebe/clients/go/internal/utils/structmap.go:80 +0x2f4
```

</p>
</details>

--- a/clients/go/internal/utils/structmap.go
+++ b/clients/go/internal/utils/structmap.go
@@ -38,6 +38,11 @@ func MapMarshal(iface interface{}, tag string, omitempty, omitminus bool) map[st
 
 func MapStructMarshal(value reflect.Value, tag string, omitempty, omitminus bool) map[string]interface{} {
 	m := make(map[string]interface{})
+
+	if !value.IsValid() {
+		return m
+	}
+
 	typeof := value.Type()
 	for i := 0; i < typeof.NumField(); i++ {
 		fieldType := typeof.Field(i)

--- a/clients/go/internal/utils/structmap_test.go
+++ b/clients/go/internal/utils/structmap_test.go
@@ -294,6 +294,27 @@ func TestMapStructMarshal(t *testing.T) {
 	if res["C"].(map[string]interface{})["Foo"].(string) != "bar" {
 		t.Fatal("map nested map string fail")
 	}
+
+}
+
+func TestMapStructMarshalWithNil(t *testing.T) {
+	res := MapStructMarshal(reflect.ValueOf(struct {
+		A *struct{}
+	}{}), "json", true, true)
+
+	if m := res["A"].(map[string]interface{}); len(m) != 0 {
+		t.Fatal("unexpected value instead of empty map")
+	}
+}
+
+func TestMapMapMarshalWithNil(t *testing.T) {
+	res := MapStructMarshal(reflect.ValueOf(struct {
+		A map[string]interface{}
+	}{}), "json", true, true)
+
+	if m := res["A"].(map[string]interface{}); len(m) != 0 {
+		t.Fatal("unexpected value instead of empty map")
+	}
 }
 
 func TestMapStructMarshal_Tagged(t *testing.T) {

